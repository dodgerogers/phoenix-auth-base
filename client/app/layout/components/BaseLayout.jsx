import React from 'react';
import { connect } from 'react-redux';
import functional from 'react-functional';
import { initialize } from 'redux-oauth';
import NavbarContainer from '../containers/NavbarContainer';
import ModalsContainer from '../../common/modals/containers/ModalsContainer';
import Footer from './Footer';
import { reduxOauthConfig } from '../constants/reduxOauth';


const BaseLayout = (props) => (
  <div className="BaseLayout">
    <NavbarContainer {...props} />
    <div className="main">
      {props.children}
    </div>
    <Footer {...props} />
    <ModalsContainer />
  </div>
);

const options = {
  componentWillMount: (props) => props.initialize(reduxOauthConfig),
}

export { BaseLayout as PureComponent };
export const FunctionalComponent = functional(BaseLayout, options);
export default connect(null, { initialize })(FunctionalComponent);
