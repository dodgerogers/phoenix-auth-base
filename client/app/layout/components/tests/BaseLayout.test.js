import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';
import toJson from 'enzyme-to-json';
import { PureComponent, FunctionalComponent } from '../BaseLayout';


describe('BaseLayout', () => {
  describe('PureComponent', () => {
    const pureComponent = (props) => shallow(
      <PureComponent {...props}>
        <div>content</div>
      </PureComponent>
    );

    it('matches snapshot', () => {
      const wrapper = pureComponent({});

      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  describe('FunctionalComponent', () => {
    const mockInitialize = jest.fn(() => true);
    const functionalComponent = (props) => shallow(
      <FunctionalComponent {...props} />, { lifecycleExperimental: true }
    );

    it('calls initialize when on componentWillMount', () => {
      const wrapper = functionalComponent({ initialize: mockInitialize });

      expect(mockInitialize).toHaveBeenCalled();
    });
  })
});
