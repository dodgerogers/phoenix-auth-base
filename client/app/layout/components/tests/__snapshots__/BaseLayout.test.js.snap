// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseLayout PureComponent matches snapshot 1`] = `
<div
  className="BaseLayout"
>
  <Connect(Connect(Wrapper))>
    <div>
      content
    </div>
  </Connect(Connect(Wrapper))>
  <GlobalNotifications />
  <div
    className="main"
  >
    <div>
      content
    </div>
  </div>
  <Footer>
    <div>
      content
    </div>
  </Footer>
  <Connect(ModalsContainer) />
</div>
`;
